L1Trigger/L1TGlobal/src/ZdcEnergySumCondition.cc

```diff
@@ -0,0 +1,268 @@
+  // energy sums are global quantities - one object per event
+
+  m_condMaxNumberObjects = 1;
+}
+
+// copy constructor
+void l1t::ZdcEnergySumCondition::copy(const l1t::ZdcEnergySumCondition& cp) {
+  m_gtZdcEnergySumTemplate = cp.gtZdcEnergySumTemplate();
+  m_uGtB = cp.getuGtB();
+
+  m_condMaxNumberObjects = cp.condMaxNumberObjects();
+  m_condLastResult = cp.condLastResult();
+  m_combinationsInCond = cp.getCombinationsInCond();
+
+  m_verbosity = cp.m_verbosity;
+}
+
+l1t::ZdcEnergySumCondition::ZdcEnergySumCondition(const l1t::ZdcEnergySumCondition& cp) : ConditionEvaluation() {
+  copy(cp);
+}
+
+// destructor
+l1t::ZdcEnergySumCondition::~ZdcEnergySumCondition() {
+  // empty
+}
+
+// equal operator
+l1t::ZdcEnergySumCondition& l1t::ZdcEnergySumCondition::operator=(const l1t::ZdcEnergySumCondition& cp) {
+  copy(cp);
+  return *this;
+}
+
+// methods
+void l1t::ZdcEnergySumCondition::setGtZdcEnergySumTemplate(const ZdcEnergySumTemplate* eSumTempl) {
+  m_gtZdcEnergySumTemplate = eSumTempl;
+}
+
+// set the pointer to uGT GlobalBoard
+void l1t::ZdcEnergySumCondition::setuGtB(const GlobalBoard* ptrGTB) { m_uGtB = ptrGTB; }
+
+// try all object permutations and check spatial correlations, if required
+const bool l1t::ZdcEnergySumCondition::evaluateCondition(const int bxEval) const {
+  // number of trigger objects in the condition
+  // in fact, there is only one object
+  int iCondition = 0;
+
+  // condition result condResult set to true if the energy sum
+  // passes all requirements
+  bool condResult = false;
+
+  // store the indices of the calorimeter objects
+  // from the combination evaluated in the condition
+  SingleCombInCond objectsInComb;
+
+  // clear the m_combinationsInCond vector
+  (combinationsInCond()).clear();
+
+  // clear the indices in the combination
+  objectsInComb.clear();
+
+  const BXVector<const l1t::EtSum*>* candVecZdc = m_uGtB->getCandL1ZdcEtSum();
+
+  // Look at objects in bx = bx + relativeBx
+  int useBx = bxEval + m_gtZdcEnergySumTemplate->condRelativeBx();
+
+  // Fail condition if attempting to get Bx outside of range
+  if ((useBx < candVecZdc->getFirstBX()) || (useBx > candVecZdc->getLastBX())) {
+    return false;
+  }
+
+  // If no candidates, no use looking any further.
+  int numberObjectsZdc = candVecZdc->size(useBx);
+
+  if (numberObjectsZdc < 1) {
+    return false;
+  }
+
+  l1t::EtSum::EtSumType type;
+  switch ((m_gtZdcEnergySumTemplate->objectType())[0]) {
+    case gtETM:
+      type = l1t::EtSum::EtSumType::kMissingEt;
+      break;
+    case gtETT:
+      type = l1t::EtSum::EtSumType::kTotalEt;
+      break;
+    case gtETTem:
+      type = l1t::EtSum::EtSumType::kTotalEtEm;
+      break;
+    case gtHTM:
+      type = l1t::EtSum::EtSumType::kMissingHt;
+      break;
+    case gtHTT:
+      type = l1t::EtSum::EtSumType::kTotalHt;
+      break;
+    case gtETMHF:
+      type = l1t::EtSum::EtSumType::kMissingEtHF;
```

Comment:
I do not understand: if this is a ZDC-related object, why does it contain all these categories, including HF-related ones ?

Reply:
I kept them because I was having problems without checking all the cases in the [DataFormat GlobalObject enum](https://github.com/cms-sw/cmssw/blob/c69c0932fceca5805b88364d7db3520338fb980f/DataFormats/L1TGlobal/interface/GlobalObject.h#L26)..

Reply:
~~Does it mean one should add handling of the ZDC types to the original `EnergySumCondition` class?~~

Ah, I see that indeed they have been added.

Maybe a silly question then - is this class needed at all?
Couldn't we just extend the original `EnergySumCondition` class to handle the ZDC case?

Reply:
Hi Andrea, not a silly question, we indeed starting with an implementation done in the EnergySumCondition code and reusing that template. However, it was then discussed to have the [GlobalBoard](https://github.com/cms-sw/cmssw/blob/2a6a807e40228bb536063ccaad2392dd3e803baa/L1Trigger/L1TGlobal/src/GlobalBoard.cc#L144) receiving ZDC signals as a completely different subsystem wrt to CaloLayer2 with [its proper inputCollection from the ZDC producer](https://github.com/cms-sw/cmssw/blob/2a6a807e40228bb536063ccaad2392dd3e803baa/L1Trigger/L1TGlobal/python/simGtStage2Digis_cfi.py). Maybe @bundocka wants to correct me or comment more here. This approach requires a [dedicated template](https://github.com/cms-sw/cmssw/blob/2a6a807e40228bb536063ccaad2392dd3e803baa/L1Trigger/L1TGlobal/interface/ZdcEnergySumTemplate.h), and, in my understanding of the GT code, a dedicated ConditionClass too.